.github/workflows/platform-validation.yml

```diff
@@ -87,3 +87,22 @@ jobs:
         run: |-
           cd example/ios
           xcodebuild build -workspace example.xcworkspace -scheme example -destination 'platform=iOS Simulator,name=iPhone 12 Pro'
+  check-typescript:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+      - name: Cache JS Dependencies
+        uses: c-hive/gha-yarn-cache@v1
+      - name: Install Yarn
+        run: npm install --global yarn
+      - name: Install Library Dependencies
+        run: yarn install
+      - name: Check Lint & Format
+        run: |
+          npm run lint:ci
+          npm run format:ci
```

Comment:
`format` is usually used for fixing the linting issues right? will the workflow then try to commit them?

Reply:
I think , it depends on how the project define `lint` and `format`.

RNTP's package.json define like this.
```
"lint": "eslint \"src/**/*.ts\"",
"format": "prettier --write \"src/**/*.ts\"",
```

Eslint fixing and Prettier fixing are not always same, so I thought it's better to check both of them.

I added `lint:ci` and `format:ci` to package.json, those command just check code, don't commit any diff.